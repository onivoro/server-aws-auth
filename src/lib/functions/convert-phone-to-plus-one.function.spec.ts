import { convertPhoneToPlusOne } from './convert-phone-to-plus-one.function';

describe('convertObjectToAttributes', () => {
    it.each([
        null, undefined, '', 0
    ])('returns empty string for falsey values like => %j', (input) => {
        expect(convertPhoneToPlusOne(input as any)).toEqual('');
    });

    it.each([
        '+17245459294', '724-545-9294', '7245459294', '(724) 545-9294', '+1(724) 545-9294'
    ])('removes dashes and prepends +1, given %j', (input) => {
        expect(convertPhoneToPlusOne(input as any)).toMatchSnapshot()
    });
});