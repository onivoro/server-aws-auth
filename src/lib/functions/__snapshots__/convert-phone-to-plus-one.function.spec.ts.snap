// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`convertObjectToAttributes removes dashes and prepends +1, given "(724) 545-9294" 1`] = `"+17245459294"`;

exports[`convertObjectToAttributes removes dashes and prepends +1, given "+1(724) 545-9294" 1`] = `"+17245459294"`;

exports[`convertObjectToAttributes removes dashes and prepends +1, given "+17245459294" 1`] = `"+17245459294"`;

exports[`convertObjectToAttributes removes dashes and prepends +1, given "724-545-9294" 1`] = `"+17245459294"`;

exports[`convertObjectToAttributes removes dashes and prepends +1, given "7245459294" 1`] = `"+17245459294"`;
